core/src/contract_env.rs

```diff
@@ -21,13 +25,14 @@ impl ContractEnv {
         }
     }
 
-    pub(crate) fn current_key(&self) -> Vec<u8> {
-        let index_bytes = key_maker::u32_to_hex(self.index);
-        let mapping_data_bytes = key_maker::bytes_to_hex(&self.mapping_data);
-        let mut key = Vec::new();
-        key.extend_from_slice(&index_bytes);
-        key.extend_from_slice(&mapping_data_bytes);
-        key
+    pub(crate) fn current_key(&self) -> StorageKey {
+        let mut result = [0u8; KEY_LEN];
```

Comment:
```suggestion
        let mut result = StorageKey::default();
```

Reply:
Sadly, arrays don't have the `default` function